### ISSUE TITLE
Web App running in IIS does not properly load credentials for AmazonSQSClient

### ISSUE THREAD

**Issue by mclark1129** [aws/aws-sdk-net#1139]
When a web application is running in IIS, the SDK is unable to load the appropriate credentials stored in the local credential file.  As a result it is falling through to default EC2 instance credentials, which obviously do not work on local developer machines.  The same code works fine when the application is running in IIS Express, or as a console application.  

I would suspect that the problem is the fact that the IIS website is running as the Application Pool identity, and the credentials configured on the command line are stored in my local user profile folder.  However, we've tried a few workarounds that in theory should be able to address that with no success.

Attempted workarounds
1. Define `AWS_SHARED_CREDENTIALS_FILE` as a system environment variable, configured to point to `C:\Users\myuseraccount\.aws\credentials`
2. Grant `Read, Read & Execute, List folder contents` permissions to the `.aws` folder to the `IIS_USRS` group on my machine

I've also attempted to configure the app pool for the website to run as my local login, but this does not seem to have any effect.

## Expected Behavior

Websites running in IIS (not IIS express) should load the correct credentials stored in the local credential file.

## Current Behavior

Websites  running in IIS (not IIS express) currently fall through to `DefaultInstanceProfileAWSCredentials`

## Steps to Reproduce (for bugs)

Sample Project: https://github.com/mclark1129/aws-credential-bug-repro

1. Initialize a default profile by using `aws configure` in the AWS CLI.
2. Configure the attached website as a new site in IIS (not IIS Express)
3. Perform a GET to `http://yourconfiguredwebsite/repro` in order to execute the sample code.  This attempts to list queues available to the profile configured via the CLI

## Context

We're attempting to migrate application code which leverages the SDK to use credentials installed on the machine (credential file for local development, EC2 instance profiles for production).  As a result we're relying on the SDK to load the appropriate credential chain depending on the machine in which the code is running.

## Your Environment

* AWSSDK.Core version used: 3.3.29.13
* Service assembly and version used: SQS 3.3.3.39
* Operating System and version: Windows 10
* Visual Studio version: 15.9.3
* Targeted .NET platform: .NET 4.6.1

[logging]: http://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-other.html

**Comment by normj** [comment]
You are on the right track. Since the app is running under the identity of the app pool the SDK is not going to look into your user folder to find credentials.

In the web.config file you can add add the **AWSProfileLocation** appSetting to look at a different location for the credentials file that is not tied to a specific user logon.

```xml
<configuration>
  <appSettings>
    <add key="AWSProfileName" value="development"/>
    <add key="AWSProfilesLocation" value="C:\aws_service_credentials\credentials"/>
  </appSettings>
</configuration>
```

When an application is running on an EC2 instance the credentials are usually found using EC2 instance metadata and an IAM role but I assume in your case you not running on an EC2 instance.

**Comment by mclark1129** [comment]
@normj I was hoping to avoid adding any custom settings to app.config, because it's only necessary when developers are running applications from their own machines.  We use EC2 instances for all of our test and production environments, so this is not an issue otherwise.  There are a number of reasons that this would be cumbersome to implement in our organization.  Given that you can override the default credential path for the CLI using the  `AWS_SHARED_CREDENTIALS_FILE` environment variable, it seems natural that the SDK would behave the same way.

We managed to ultimately find a solution using environment variables to store credentials.  This works best for us regardless because we require MFA on all developer accounts.  We use the credentials in our local profile to request MFA tokens, and then our applications read those from the environment variables.  

We're no longer blocked by the issue, but I'll leave it open since I think that the SDK should probably respect the path specified in `AWS_SHARED_CREDENTIALS_FILE`.  Feel free to close if you disagree or don't intend to implement that change.

**Comment by normj** [comment]
I agree the SDK should support the environment variable. I created a new issue specifically for supporting the environment variable. I'm going to close this issue and use the other one to track the feature request.

**Comment by ashclarke** [comment]
For those using `LocalSystem` as their app pool identity for local development, the `.aws` folder is expected to be found in `C:\Windows\System32\config\systemprofile\`.

Therefore, the full filepath that AWS SDK uses to find the credentials file, in this case, is:

`C:\Windows\System32\config\systemprofile\.aws\credentials`

**Comment by casey-robertson** [comment]
> @normj I was hoping to avoid adding any custom settings to app.config, because it's only necessary when developers are running applications from their own machines. We use EC2 instances for all of our test and production environments, so this is not an issue otherwise. There are a number of reasons that this would be cumbersome to implement in our organization. Given that you can override the default credential path for the CLI using the `AWS_SHARED_CREDENTIALS_FILE` environment variable, it seems natural that the SDK would behave the same way.
> 
> We managed to ultimately find a solution using environment variables to store credentials. This works best for us regardless because we require MFA on all developer accounts. We use the credentials in our local profile to request MFA tokens, and then our applications read those from the environment variables.
> 
> We're no longer blocked by the issue, but I'll leave it open since I think that the SDK should probably respect the path specified in `AWS_SHARED_CREDENTIALS_FILE`. Feel free to close if you disagree or don't intend to implement that change.

If you have any details on the solution I'd love to hear about them.  Moving workload to EKS/AWS and this is an issue we are discussing since the past we have used JSON credential files mounted as K8s secrets.

**Comment by imaramos** [comment]
Hi

Do we have any news on this? Do we have any other to accomplish it besides put it in web.config?
This issue https://github.com/aws/aws-sdk-net/issues/1158 is still open.
Thank you

**Comment by kshitijsharmanet** [comment]
Hello, I am having same issue. Did anyone found a solution yet?

**Comment by MCHouston** [comment]
Hey there,

Just wanted to hop in and say that I was able to get this resolved on my end using the suggestion from @ashclarke, but I also had to give full permissions to IIS_IUSRS on the credentials file to get it to read correctly.

**Comment by Rahulrwts** [comment]
I am getting same issue is there any resolution
